main.go

```diff
@@ -28,6 +28,7 @@ import (
 	// +kubebuilder:scaffold:imports
 	aadpodv1 "github.com/Azure/aad-pod-identity/pkg/apis/aadpodidentity/v1"
 	asocontainerservicev1 "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230201"
+	asocontainerservicev1preview "github.com/Azure/azure-service-operator/v2/api/containerservice/v1api20230315preview"
```

Comment:
I think it would be wise to bump ASO to use the latest API, and get CAPZ to use the new version before merging this.

Reply:
added here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4316/commits/b354f65099b9cd73b66b7a090c9844affed3a5d1#diff-4297dcd4bfa35df2d292734776d86f714d53c30a13aa2fc5795cb549a6f03152L37